great_expectations/data_context/types/base.py

```diff
@@ -1405,10 +1405,38 @@ def validate_schema(self, data, **kwargs):
             "name" in data or "validation_operator_name" in data or "batches" in data
         ):
             raise ge_exceptions.InvalidConfigError(
-                f"""Your current Checkpoint configuration is incomplete.  Please update your configuration to continue.
+                f"""Your current Checkpoint configuration is incomplete.  Please update your checkpoint configuration to
+                continue.
                 """
             )
 
+        if data.get("config_version"):
+            if "name" not in data:
+                raise ge_exceptions.InvalidConfigError(
+                    f"""Your Checkpoint configuration requires the "name" field.  Please update your current checkpoint
+                    configuration to continue.
+                    """
+                )
+            action_list: Optional[list] = data.get("action_list")
+            action_list_present: bool = (
+                action_list is not None
+                and isinstance(action_list, list)
+                and len(action_list) > 0
+            )
+            if not (
+                (
+                    "validations" in data
+                    and isinstance(data["validations"], list)
+                    and len(data["validations"]) > 0
+                )
+                or action_list_present
+            ):
+                raise ge_exceptions.InvalidConfigError(
+                    f"""Your current Checkpoint configuration is inconsistent.  Please update your checkpoint
+                    configuration to continue.
+                    """
+                )
+
```

Comment:
Not sure if we want to do validation of these attributes here, since user should be allowed to simply pass list of validations to run at runtime. If each validation has it's own action_list, then both the top-level action list and validations keys can be empty.

Reply:
They are now a warning and part of `Checkpoint.self_check()`.